app/src/main/java/com/rage/clamber/Fragments/WallSectionFragment.java

```diff
@@ -45,23 +74,68 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         View rootView = inflater.inflate(R.layout.fragment_wall_section, container, false);
         ButterKnife.bind(this, rootView);
-
+        wallSections = new ArrayList<>();
+        mainUser = getArguments().getParcelable(HomePage.ARG_USER);
+        wall = getArguments().getInt(WallsFragment.ARG_WALL_ID);
+        getWallSectionsByWall(mainUser.getUserName(), wall);
         recyclerView.setLayoutManager(new GridLayoutManager(getActivity(), 3));
-        WallsPageRecyclerViewAdapter adapter = new WallsPageRecyclerViewAdapter(WALL_NAMES, this);
-        recyclerView.setAdapter(adapter);
-        recyclerView.setItemAnimator(new DefaultItemAnimator());
+
 
         return rootView;
     }
 
+    /**
+     * Method to make a network call to the Clamber Server to get all WallSections for the specified
+     * Wall. It updates the WallsPageRecyclerViewAdapter to display the returned WallSection data.
+     * @param userName - userName is needed for the path parameters.
+     * @param wallId - wallId tells the query which wall to pull sections from.
+     */
+    public void getWallSectionsByWall(String userName, int wallId) {
+        ConnectivityManager connectivityManager = (ConnectivityManager) getActivity().getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+        if (networkInfo != null && networkInfo.isConnected()) {
+            Retrofit retrofit = new Retrofit.Builder()
+                    .baseUrl(HomePage.CONNECTION_WEB_ADDRESS)
+                    .addConverterFactory(JacksonConverterFactory.create())
+                    .build();
+
+            ClamberService clamber = retrofit.create(ClamberService.class);
+            final Call<List<WallSection>> wallSectionCall = clamber.getWallSectionByWall(userName, wallId);
+
+            wallSectionCall.enqueue(new Callback<List<WallSection>>() {
+                @Override
+                public void onResponse(Call<List<WallSection>> call, Response<List<WallSection>> response) {
+
+                    if (response.code() == 200){
+                     List<WallSection> wallSectionsArrayList = response.body();
+                        for (int i = 0; i < wallSectionsArrayList.size(); i++){
```

Comment:
Fortunately, the standard library designers accounted for this being a common use case so you can do this in one line by saying: `wallSections.addAll(wallSectionsArrayList)`